### PR TITLE
WIP - Correct Attribute fetching

### LINKED ISSUE #888: Data tab - filter/format attributes
Data tab table should only show attributes that are allowed for layers in metadata object. Currently it shows all available attributes for any given feature.

- [x] remote data layers attribute filtering
- [x] webmap layer attribute filtering
- [ ] check for formatting options and format attribute data appropriately

### PR DESCRIPTION
**Pull request by vaidotasp**
Fixes #888

- Extracting popup information along with other metadata from the point of Layer creation. This information will be used for LegendConfig as well as Data Tab
- Types modifications to include expanded metadata objects
- Reworked DataPanel.ts, feature fetching logic for data panel:
1 . Made queries more sequential and less prone to duplicate.
2. Custom logic to fetch all available fields on the layer in order to extract OBJECTID. We will need object id when to fetch documents
3. Implemented logic where if user have no specified fields in the metadata, we return all fields. Decided to return all fields if there is no config. #895

### CODE REVIEW

File: src/js/controllers/mapController.ts
Comment (vaidotasp) [inline src/js/controllers/mapController.ts:216]: here is where we get the configs/metadata/sublabels etc etc etc

File: src/js/controllers/mapController.ts
Comment (vaidotasp) [inline src/js/controllers/mapController.ts]: ?

File: src/js/controllers/mapController.ts
Comment (vaidotasp) [inline src/js/controllers/mapController.ts]: do we need to return null for fieldNames?

**Comment by vaidotasp** [comment]
sublayerID sublayerTitle, could those be optionals instead of union between null?

**Review by eyosef — COMMENTED** [review]
@vaidotasp LGTM - let me know if you decide to implement the `interface` update, or the `util.ts` update for `DataPanel`. Then I'll do additional testing before approving and merging into `clean-slate`

**Review by eyosef — APPROVED** [review]
LGTM! Thanks for moving the helper functions into `helpers/dataPanel`!

**Pull request merged**
